StyleCop.Analyzers/StyleCop.Analyzers/Helpers/AccessLevelHelper.cs

```diff
@@ -78,5 +81,508 @@ internal static string GetName(AccessLevel accessLevel)
+                else
+                {
+                    return Accessibility.Public;
+                }
+            }
+
+            if (syntax.IsKind(SyntaxKind.ConstructorDeclaration))
+            {
+                return Accessibility.Private;
+            }
+
+            IMethodSymbol declaredSymbol = semanticModel.GetDeclaredSymbol(syntax, cancellationToken);
+            return declaredSymbol?.DeclaredAccessibility ?? Accessibility.NotApplicable;
+        }
+
+        internal static Accessibility GetDeclaredAccessibility(this BasePropertyDeclarationSyntax syntax, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (syntax == null)
+            {
+                throw new ArgumentNullException(nameof(syntax));
+            }
+
+            if (semanticModel == null)
+            {
+                throw new ArgumentNullException(nameof(semanticModel));
+            }
+
+            AccessLevel accessLevel = GetAccessLevel(syntax.Modifiers);
+            switch (accessLevel)
+            {
+            case AccessLevel.Public:
+                return Accessibility.Public;
+
+            case AccessLevel.Internal:
+                return Accessibility.Internal;
+
+            case AccessLevel.ProtectedInternal:
+                return Accessibility.ProtectedOrInternal;
+
+            case AccessLevel.Protected:
+                return Accessibility.Protected;
+
+            case AccessLevel.Private:
+                return Accessibility.Private;
+
+            case AccessLevel.NotSpecified:
+            default:
+                break;
+            }
+
+            PropertyDeclarationSyntax propertyDeclarationSyntax = syntax as PropertyDeclarationSyntax;
+            if (propertyDeclarationSyntax != null)
+            {
+                if (propertyDeclarationSyntax.ExplicitInterfaceSpecifier == null)
+                {
+                    return Accessibility.Private;
+                }
+                else
+                {
+                    return Accessibility.Public;
+                }
+            }
+
+            IndexerDeclarationSyntax indexerDeclarationSyntax = syntax as IndexerDeclarationSyntax;
+            if (indexerDeclarationSyntax != null)
+            {
+                if (indexerDeclarationSyntax.ExplicitInterfaceSpecifier == null)
+                {
+                    return Accessibility.Private;
+                }
+                else
+                {
+                    return Accessibility.Public;
+                }
+            }
+
+            EventDeclarationSyntax eventDeclarationSyntax = syntax as EventDeclarationSyntax;
+            if (eventDeclarationSyntax != null)
+            {
+                if (eventDeclarationSyntax.ExplicitInterfaceSpecifier == null)
+                {
+                    return Accessibility.Private;
+                }
+                else
+                {
+                    return Accessibility.Public;
```

Comment:
For documentation purposes, the declared accessibility of such a method would match the effective accessibility of the declaring interface, but that behavior is not currently implemented.